### ISSUE TITLE
Connect to MySQL with require_secure_transport = ON

### ISSUE THREAD

**Issue by jiyosub** [nocodb/nocodb#425]
Node: **v15.12.0**
Arch: **x64**
Platform: **darwin**
Docker: **false**
Database: **mysql2**
ProjectOnRootDB: **true**
RootDB: **mysql2**
PackageVersion: **0.11.8**
**Expected behavior**

Currently using MaraDB in docker.  Image ID: fd17f5776802.   We would like to use SSL to connect to the DB without acertificate verification.  Currently there is no way to do this using the NC_DB variable.  Technically, you should be able to add NODE_TLS_REJECT_UNAUTHORIZED = "0" to the environment, and ssl would work.  But  this throws: TypeError: SSL profile must be an object, instead it's a boolean.  This is a bug in the MySQL Driver.

What is needed, is to create this object and pass it to KNEX config:

const knex = require('knex')({
  client: 'mysql2',
  connection: {
    user: 'root',
    password: 'pass',
    host: 'localhost',
      database: 'db',
    ssl: {rejectUnauthorized: false}
  }
});

The end result is being able to connect to mysql with require_secure_transport = ON.
 
[my.cnf.zip](https://github.com/nocodb/nocodb/files/6930528/my.cnf.zip)

**Comment by o1lab** [comment]
We allow to change knex connection config - see below.

https://www.youtube.com/watch?v=rS0NRwRCjx0&t=2s

**Comment by jiyosub** [comment]
This works with some changes. I had to remove the env variable NC_DB and edited the nc_store.json file directly.

```
[
  {
    "id": 4,
    "project_id": "big_brain_Dw1h",
    "db_alias": "db",
    "key": "NC_CONFIG",
    "value": "{\"version\":\"0009044\",\"queriesFolder\":\"\",\"meta\":{\"db\":{\"client\":\"mysql\",\"connection\":{\"database\":\"mydb\",\"host\":\"localhost\",\"password\":\"mypass\",\"port\":3306,\"user\":\"root\",\"ssl\":{\"rejectUnauthorized\": \"false\"}},\"acquireConnectionTimeout\":600000}},\"try\":false,\"dashboardPath\":\"/dashboard\",\"auth\":{\"jwt\":{\"dbAlias\":\"db\"}},\"port\":8080,\"env\":\"dev\",\"workingEnv\":\"dev\",\"toolDir\":\"/Users/joseph/Dev/nocodb/packages/nocodb\",\"projectType\":\"rest\",\"type\":\"docker\",\"title\":\"Big Brain\",\"prefix\":\"xa1__\",\"id\":\"big_brain_Dw1h\"}",
    "type": null,
    "env": null,
    "tag": null,
    "created_at": "2021-08-03T18:11:21.000Z",
    "updated_at": "2021-08-03T18:11:21.000Z"
  }
]
```

Was this documented somewhere and I missed it?

**Comment by jiyosub** [comment]
There might be a bit of a catch 22.  You need a DB connection to create the JSON config.  But you can't create a connection with NC_DB.  So I had to turn TLS off in the db, connect, create the JSON config, then turn it back on.

**Comment by pranavxc** [comment]
> NC_DB

MAny

> There might be a bit of a catch 22. You need a DB connection to create the JSON config. But you can't create a connection with NC_DB. So I had to turn TLS off in the db, connect, create the JSON config, then turn it back on.

Maybe we will add something like  `NC_DB_JSON` environment variable which accepts knex connection config. Hope that will solve the problem.

**Comment by jiyosub** [comment]
Personally, I'd like to see the DB for the dashboard be completely separate from project DBs.  Two different  structures:
NC_DASH_DB_JSON
NC_PROJ_DB_JSON

**Comment by pranavxc** [comment]
> Personally, I'd like to see the DB for the dashboard be completely separate from project DBs. Two different structures:
> NC_DASH_DB_JSON
> NC_PROJ_DB_JSON

If you want to create a project in external DB, you can click on the `Create By Connecting External DB` option.

**Comment by jiyosub** [comment]
This is true, but then you have credentials stored in different places.  And it also means, sometimes project databases are stored in the same DB as the dashboard (if you don't supply an external DB string).  Personally, I'd silo it a bit more.  Have a separate dashboard DB, then the ability to crate new projects either in the same server (different DB) or provide an external DB.

**Comment by pranavxc** [comment]
You can add any additional parameter in the connection URL, for the nested property you can use dot(`.`) to represent the nested property.  

For example `ssl: {rejectUnauthorized:false}` can be added by specifying `ssl.rejectUnauthorized=false`. 


Sample url 

```
pg://localhost:5432?u=postgres&p=xgene&d=abcde&ssl.rejectUnauthorized=false
```

Available since `0.11.16`
